baseband/vlbi_base/frame.py

```diff
@@ -122,10 +122,20 @@ def fromdata(cls, data, header, *args, **kwargs):
         payload = cls._payload_class.fromdata(data, *args, **kwargs)
         return cls(header, payload, valid=valid, verify=verify)
 
+    @property
+    def sample_shape(self):
+        """Shape of the samples held in the frame (nchan,)."""
+        return self.payload.sample_shape
+
+    @property
+    def samples_per_frame(self):
```

Comment:
I see the dilemma between `samples_per_frame` and `nsample` here.  I have a slight preference for `nsample`, since the payload technically doesn't know it's in a frame, and the frame doesn't know it's in a stream.

Reply:
I think I'll go for removing both and instead relying on `__len__`.